File: apps/android/feature/profile/ui/src/main/java/com/scribblefit/feature/profile/ui/ProfileViewModel.kt
package com.scribblefit.feature.profile.ui

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.scribblefit.feature.profile.domain.model.UserStats
import com.scribblefit.feature.profile.domain.repository.UserRepository
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.SharingStarted
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.map
import kotlinx.coroutines.flow.stateIn
import javax.inject.Inject

private const val FLOW_TIMEOUT_MS = 5_000L

data class ProfileUiState(
    val userName: String = "George",
    val stats: UserStats? = null,
    val isLoading: Boolean = true
)

@HiltViewModel
class ProfileViewModel @Inject constructor(
    private val userRepository: UserRepository,
    private val navigator: com.scribblefit.core.navigation.Navigator
) : ViewModel() {

    val uiState: StateFlow<ProfileUiState> = userRepository.getUserStats()
        .map { stats ->
            ProfileUiState(stats = stats, isLoading = false)
        }
        .stateIn(
            scope = viewModelScope,
            started = SharingStarted.WhileSubscribed(FLOW_TIMEOUT_MS),
            initialValue = ProfileUiState()
        )

    fun onSettingsClick() {
        navigator.navigateTo(com.scribblefit.core.navigation.Screen.Settings)
    }
}

File: apps/ios/ScribbleFit/ScribbleFit/Features/Profile/UI/ProfileViewModel.swift
import Foundation
import SwiftUI
import Combine

@MainActor
public final class ProfileViewModel: ObservableObject {
    private let userRepository: UserRepository
    private let navManager: NavigationManager
    
    @Published public var uiState = ProfileUiState()
    
    public init(userRepository: UserRepository, navManager: NavigationManager) {
        self.userRepository = userRepository
        self.navManager = navManager
        
        refreshStats()
    }
    
    public func refreshStats() {
        Task {
            do {
                let stats = try await userRepository.getUserStats()
                self.uiState = ProfileUiState(userName: "George", stats: stats, isLoading: false)
            } catch {
                print("Failed to fetch user stats: \(error)")
                self.uiState.isLoading = false
            }
        }
    }
    
    public func onSettingsClick() {
        navManager.navigate(to: AppDestination.settings, in: .profile)
    }
}

public struct ProfileUiState {
    public var userName: String = "George"
    public var stats: UserStats? = nil
    public var isLoading: Bool = true
    
    public init(userName: String = "George", stats: UserStats? = nil, isLoading: Bool = true) {
        self.userName = userName
        self.stats = stats
        self.isLoading = isLoading
    }
}
